src/containers/Application/ApplicationPage.tsx

```diff
@@ -2,32 +2,96 @@ import React, { useEffect, useState } from 'react'
 import { FullStructure, ResponsesByCode, ElementStateNEW } from '../../utils/types'
 import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
 import { ApplicationStatus } from '../../utils/generated/graphql'
-import { useApplicationState } from '../../contexts/ApplicationState'
 import { useUserState } from '../../contexts/UserState'
 import { useRouter } from '../../utils/hooks/useRouter'
-import { Loading, NoMatch } from '../../components'
+import { Loading } from '../../components'
 import strings from '../../utils/constants'
-import messages from '../../utils/messages'
 import { Button, Grid, Header, Message, Segment, Sticky } from 'semantic-ui-react'
 import { PageElements } from '../../components/Application'
+import { useFormElementUpdateTracker } from '../../contexts/FormElementUpdateTrackerState'
 
 interface ApplicationProps {
   structure: FullStructure
   responses?: ResponsesByCode
 }
+// TODO move to shared types if used elsewhere
+type SectionAndPage = { sectionCode: string; pageName: string } | null
+
+interface MethodToCallOnRevalidation {
+  (firstInvalidPage: SectionAndPage): void
+}
```

Comment:
Also this one would be used by ProgressBar - See my TODO on PR #373

Reply:
moved to types: 1b280a6189c93cfc82302916d5cfd682866f817c